projects/fsastorefrontlib/src/cms-components/seller-dashboard/seller-dashboard-list.guard.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from '@angular/core';
+import { CanActivate } from '@angular/router';
+import {
+  GlobalMessageService,
+  GlobalMessageType,
+  RoutingService,
+  User,
+} from '@spartacus/core';
+import { UserAccountFacade } from '@spartacus/user/account/root';
+import { Observable } from 'rxjs';
+import { filter, map, pluck } from 'rxjs/operators';
+import { FSUserRole } from '../../occ/occ-models/occ.models';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class SellerDashboardListGuard implements CanActivate {
+  constructor(
+    protected userAccountFacade: UserAccountFacade,
+    protected routingService: RoutingService,
+    protected globalMessageService: GlobalMessageService
+  ) {}
+
+  canActivate(): Observable<boolean> {
+    return this.userAccountFacade.get().pipe(
+      filter((user: User) => user && Object.keys(user).length > 0),
+      pluck('roles'),
+      map((roles: string[]) => {
+        const hasRole =
+          Array.isArray(roles) && roles.includes(FSUserRole.SELLER);
```

Comment:
roles is already of type string[], is this Array.isArray(roles) needed than bcs type is already matching